API/Utilities/ClaimsPrincipalExtensions.cs

```diff
@@ -43,6 +41,34 @@ public static bool IsUser(this ClaimsPrincipal claimsPrincipal)
     public static bool IsMatchVerifier(this ClaimsPrincipal claimsPrincipal) =>
         IsInRole(claimsPrincipal, "verifier");
 
+    /// <summary>
+    /// Denotes the principle as having either the verifier or admin role
+    /// </summary>
+    /// <param name="claimsPrincipal"></param>
+    /// <returns></returns>
+    public static bool CanVerifyMatches(this ClaimsPrincipal claimsPrincipal) =>
+        (claimsPrincipal.IsAdmin() || claimsPrincipal.IsMatchVerifier());
+
+    /// <summary>
+    /// Returns the appropriate <see cref="MatchVerificationSource"/> enum for the principle
+    /// </summary>
+    /// <param name="claimsPrincipal"></param>
+    /// <returns></returns>
+    public static MatchVerificationSource? VerificationSource(this ClaimsPrincipal claimsPrincipal)
+    {
+        if (claimsPrincipal.IsAdmin())
+        {
+            return MatchVerificationSource.Admin;
```

Comment:
General comment, eventually I want to get rid of this distinction. It makes no difference to us.